src/BaseActions.tsx

```diff
@@ -19,8 +19,20 @@ export function goBack(): Action {
   return { type: 'GO_BACK' };
 }
 
-export function navigate(name: string, params?: object): Action {
-  return { type: 'NAVIGATE', payload: { name, params } };
+export function navigate(target: TargetForAction, params?: object): Action {
+  if (
+    (target.hasOwnProperty('key') && target.hasOwnProperty('name')) ||
+    (!target.hasOwnProperty('key') && !target.hasOwnProperty('name'))
+  ) {
+    throw new Error(
```

Comment:
Maybe instead of throwing in the action creator, we can throw in the router. Then it'll also prevent someone from calling `dispatch` directly with incorrect stuff

Reply:
I don't think so. We don't want anyone to dispatch `NAVIGATE` directly. If yes, risk included. I don't want to copy this error to every new and existing navigators.